daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/NormalizeRollbacks.scala

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package speedy
+
+import com.daml.lf.transaction.{NodeId, GenTransaction}
+import com.daml.lf.transaction.Node.{GenNode, NodeRollback, NodeExercises, LeafOnlyActionNode}
+import com.daml.lf.value.Value
+import com.daml.lf.data.ImmArray
+
+import scala.annotation.tailrec
+
+private[lf] object NormalizeRollbacks {
+
+  type Nid = NodeId
+  type Cid = Value.ContractId
+  type TX = GenTransaction[Nid, Cid]
+  type Node = GenNode[Nid, Cid]
+  type LeafNode = LeafOnlyActionNode[Cid]
+  type ExeNode = NodeExercises[Nid, Cid]
```

Comment:
Why this is not private ? 

```suggestion
  private[this] type Nid = NodeId
  private[this]  type Cid = Value.ContractId
  private[this] type TX = GenTransaction[Nid, Cid]
  private[this] type Node = GenNode[Nid, Cid]
  private[this] type LeafNode = LeafOnlyActionNode[Cid]
  private[this] type ExeNode = NodeExercises[Nid, Cid]
```

Reply:
done